SpatialGDK/Source/SpatialGDK/Public/EngineClasses/Components/SpatialPingComponent.h

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#pragma once
+
+#include "CoreMinimal.h"
+#include "Components/ActorComponent.h"
+
+#include "SpatialPingComponent.generated.h"
+
+/*
+ Offers a configurable means of measuring round-trip latency in SpatialOS deployments.
+ This component should be attached to a player controller.
+ */
+UCLASS(ClassGroup = (SpatialGDK), Meta = (BlueprintSpawnableComponent))
+class SPATIALGDK_API USpatialPingComponent : public UActorComponent
+{
+	GENERATED_UCLASS_BODY()
+
+public:
+	UPROPERTY(EditAnywhere, BlueprintReadWrite, Category = SpatialPing)
+	bool bStartWithPingEnabled = true;
+	UPROPERTY(EditAnywhere, BlueprintReadWrite, Category = SpatialPing)
+	float PingFrequency = 0.5f;
+
+	virtual void BeginPlay() override;
+	virtual void EndPlay(const EEndPlayReason::Type EndPlayReason) override;
+
+	//Returns whether this component can ping
+	UFUNCTION(BlueprintCallable, Category = "SpatialGDK|Ping")
+	bool GetIsPingEnabled() const;
+
+	//Set whether this component can ping
+	UFUNCTION(BlueprintCallable, Category = "SpatialGDK|Ping")
+	void SetPingEnabled(bool bSetEnabled);
+
+	//Returns the latest raw round-trip ping value in seconds
+	UFUNCTION(BlueprintCallable, Category = "SpatialGDK|Ping")
+	float GetPing() const;
+
+private:
+	float RTPing;
+	float LastReceivedPingID;
```

Comment:
It's a bit confusing that this is an ID and timestamp at the same time. Can we just go with `LastReceivedPingTimestamp` instead? (also in other places with ID)

Reply:
Yeah, I originally started with a paired ID and timestamp and in my head the name still made sense. Not very descriptive though, you're right. Done.